pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,192 @@
+2. You have a Kubernetes cluster authenticated to the Docker registry
+4. Agents have `kubectl` installed
+3. Agents have `kubectl` access to the Kubernetes cluster
+
+<section class="Docs__note">
+  <p>
+    This setup is outside this tutorial's scope. However, you may refer to the
+    <a href="#notes">notes</a> for high level advice.
+  </p>
+</section>
+
+## Create the Deploy Pipeline
+
+Buildkite supports multiple pipelines per project. The best practice is to use
+separate pipelines for testing and deploying.  Deploy pipelines are just
+like other pipelines except they're triggered by different events. You can use
+the [Github Deployment API](https://buildkite.com/blog/github-deployments) or
+[trigger steps](https://buildkite.com/docs/pipelines/trigger-step).
+
+Start by [creating a new pipline](https://buildkite.com/docs/pipelines/defining-steps)
+that uploads steps from `.buildkite/pipeline.deploy.yml`.
+
+<%= image "new_pipeline.png" %>
+
+Now, configure the Github webhook to send deployment events.
+
+<%= image "pipeline_webhook.png" %>
+
+Next, open the pipeline settings in Buildkite. Select "Trigger builds on deployments."
+
+<%= image "pipeline_settings.png" %>
+
+Finally, create `.buildkite/pipeline.deploy.yml`. A single step will do for now:
+
+```yml
+steps:
+  - label: "\:rocket\: Push to \:kubernetes\:"
+    comamnd: script/buildlkite/deploy
+    concurrency: 1
+    concurrency_group: deploy/tutorial
+```
+
+Set `concurrency` and `concurrency_group` when updating mutable state. These
+settings ensures only one step runs at a time, effectively creating a deploy
+queue.
+
+Now the pipeline is wired up. Time for the deploy step.
+
+## Setup the Deploy Step
+
+Create a Kubernetes deployment manifest that references the `DOCKER_IMAGE`
+environment variable. `kubectl` does not interpolate environment variables out
+of the box, so you'll need a different solution. The `envsubst` command works
+well for this case. `envsubst` ("environment substitute") replaces environment
+variables from standard input. The result may be piped directly to `kubectl`.
+
+First, You'll need a deployment YML file. Call this file `k8s/deployment.yml`:
+
+```yml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tutorial
+  labels:
+    app: tutorial
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: tutorial
+  template:
+    metadata:
+      labels:
+        app: tutorial
+    spec:
+      containers:
+        - name: app
+          image: "${DOCKER_IMAGE}"
+          ports:
+            - containerPort: 3000
+```
+
+<section class="Docs__note">
+  <p>
+    Please refer to the official Kubernetes <a
+    href="https://kubernetes.io/docs/concepts/workloads/controllers/deployment/">deployment
+    documentation</a> for more information. Also note that there are multiple
+    ways to deploy applications to Kubernetes.
+  </p>
+</section>
+
+Now create a short deploy script (`script/buildkite/deploy`):
```

Comment:
What does this deploy script do? Same as above, just something quick about 'in the below example code, we check to see if the docker image is available and then run the kube deploy using kubectl' or some such.